src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,15 +1,273 @@
+      hasImdbIdError: false,
+    });
+  };
+
+  isValidTitle = () => {
+    const { title } = this.state;
+
+    if (!title) {
+      this.setState({ hasTitleError: !title });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  linkCheck = (url: string) => {
+    return url.match(/^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/);
+  };
+
+  isValidImgUrl = () => {
+    const { imgUrl } = this.state;
+    const correctImgUrl = imgUrl && this.linkCheck(imgUrl);
+
+    if (!correctImgUrl) {
+      this.setState({ hasImgUrlError: !correctImgUrl });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  isValidImdbUrl = () => {
+    const { imdbUrl } = this.state;
+    const correctImdbUrl = imdbUrl && this.linkCheck(imdbUrl);
+
+    if (!correctImdbUrl) {
+      this.setState({ hasImdbUrlError: !correctImdbUrl });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  isValidImdbId = () => {
+    const { imdbId } = this.state;
+
+    if (!imdbId) {
+      this.setState({ hasImdbIdError: !imdbId });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  validateForm = () => {
+    const {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = this.state;
+
+    if (!title || !description || !imgUrl || !imdbUrl || !imdbId) {
+      this.setState({
+        hasTitleError: !title,
+        hasImgUrlError: !imgUrl,
+        hasImdbUrlError: !imdbUrl,
+        hasImdbIdError: !imdbId,
+      });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  clearState = () => {
+    this.setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+      hasTitleError: false,
+      hasImgUrlError: false,
+      hasImdbUrlError: false,
+      hasImdbIdError: false,
+    });
+  };
+
+  handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isFormValid = this.validateForm();
+
+    if (isFormValid) {
+      const newMovie = {
+        title: this.state.title,
+        description: this.state.description,
+        imgUrl: this.state.imgUrl,
+        imdbId: this.state.imdbId,
+        imdbUrl: this.state.imdbUrl,
+      };
+
+      this.props.onAdd(newMovie);
+      this.clearState();
+    }
+  };
 
   render() {
+    const {
+      title,
+      description,
+      imdbId,
+      imdbUrl,
+      imgUrl,
+      isValidLink,
+      hasTitleError,
+      hasImgUrlError,
+      hasImdbUrlError,
+      hasImdbIdError,
+    } = this.state;
+
     return (
-      <form>
-        Put the form here
+      <form onSubmit={this.handleSubmit}>
+        <input
+          name="title"
+          type="text"
+          placeholder="title"
+          value={title}
+          className="input"
+          onChange={this.handleTitleChange}
+          onBlur={this.isValidTitle}
+        />
```

Comment:
'title, imgUrl, imdbUrl, imdbId should be required'
```suggestion
      <form onSubmit={this.handleSubmit}>
        <input
          name="title"
          type="text"
          placeholder="title"
          value={title}
          className="input"
          onChange={this.handleTitleChange}
          onBlur={this.isValidTitle}
          required
        />
```